src/basic/strv.c

```diff
@@ -353,6 +353,51 @@ int strv_split_extract(char ***t, const char *s, const char *separators, Extract
         return (int) n;
 }
 
+int strv_split_extract_pairs_colon(char ***t, const char *s) {
+        _cleanup_strv_free_ char **l = NULL;
+        size_t n = 0, allocated = 0;
+        int r;
+
+        assert(t);
+        assert(s);
+
+        for (;;) {
+                _cleanup_free_ char *first = NULL, *second = NULL, *tuple = NULL;
+                const char *p = NULL;
+
+                r = extract_first_word(&s, &tuple, WHITESPACE, EXTRACT_UNQUOTE|EXTRACT_DONT_COALESCE_SEPARATORS);
+                if (r < 0)
+                        return r;
+                if (r == 0)
+                        break;
+
+                p = tuple;
+                r = extract_many_words(&p, ":", EXTRACT_UNQUOTE|EXTRACT_DONT_COALESCE_SEPARATORS, &first, &second, NULL);
+                if (r < 0)
+                        return r;
+                if (r == 0)
+                        continue;
+
+                if (!GREEDY_REALLOC(l, allocated, n + 3))
+                        return -ENOMEM;
+
+                l[n++] = TAKE_PTR(first);
+                l[n++] = strdup(strempty(second));
```

Comment:
oom check missing here.

The caller cannot distinguish `a:a b c:c` from `a:a b: c:c`. I'm not sure if it matters anywhere?

Reply:
Added.
For the current use cases it doesn't matter, as far as I can see, as there's no particular semantic meaning associated with not specifying a second element vs explicitly setting it as empty - we can revise if this gets used elsewhere and it's needed.